pkg/armrpc/asyncoperation/controller/result.go

```diff
@@ -40,20 +44,24 @@ func NewCanceledResult(message string) Result {
 }
 
 // NewFailedResult creates a new Result object with the given error details and sets the failed flag to true.
-func NewFailedResult(err v1.ErrorDetails) Result {
+func NewFailedResult(ctx context.Context, err error, errorDetails v1.ErrorDetails) Result {
 	r := Result{}
-	r.SetFailed(err, false)
+	r.SetFailed(ctx, err, errorDetails, false)
+
 	return r
 }
 
 // SetFailed sets the error response with Failed status.
-func (r *Result) SetFailed(err v1.ErrorDetails, requeue bool) {
+func (r *Result) SetFailed(ctx context.Context, err error, errorDetails v1.ErrorDetails, requeue bool) {
+	logger := ucplog.FromContextOrDiscard(ctx)
+	logger.Error(err, fmt.Sprintf("Operation Failed: %s", errorDetails.Message))
```

Comment:
Another downside to add logging here is that whenever backend controller calls this method, it will log out. All these helpers is to construct and set the failure reason, not print out the log. adding logging here doesn't seem right..